repos/public-user/src/Module/AclRules.php

```diff
@@ -0,0 +1,181 @@
+<?php
+
+namespace PublicUser\Module;
+
+use Omeka\Entity\User;
+use Omeka\Permissions\Assertion\HasSitePermissionAssertion;
+use Omeka\Permissions\Assertion\OwnsEntityAssertion;
+use Omeka\Permissions\Assertion\SiteIsPublicAssertion;
+use Psr\Container\ContainerInterface;
+use PublicUser\Acl\IsOnRouteAssertion;
+use PublicUser\Acl\IsRegistrationPermittedAssertion;
+use PublicUser\Acl\UserHasSitePermission;
+use PublicUser\Controller\AccountController;
+use PublicUser\Controller\AuthController;
+use PublicUser\Controller\LoginController;
+use PublicUser\Controller\PublicProfileController;
+use PublicUser\Controller\SiteLoginRedirectController;
+use PublicUser\Settings\PublicUserSettings;
+use PublicUser\Site\SiteProvider;
+use Zend\Permissions\Acl\Acl;
+use Zend\Permissions\Acl\Assertion\AssertionAggregate;
+use Zend\Router\RouteInterface;
+
+trait AclRules
+{
+    /**
+     * @return ContainerInterface
+     */
+    abstract public function getServiceLocator();
+
+    private function addAclRules(RouteInterface $router, $request)
+    {
+        /** @var Acl $acl */
+        $acl = $this->getServiceLocator()->get('Omeka\Acl');
+        /** @var PublicUserSettings $settings */
+        $settings = $this->getServiceLocator()->get(PublicUserSettings::class);
+        /** @var SiteProvider $siteProvider */
+        $siteProvider = $this->getServiceLocator()->get(SiteProvider::class);
+
+        // Site permissions
+        // The configuration for the extended site permissions (what the new roles can do) is added by the PublicUser
+        // and so the definitions for the site permissions are too. This is crossing over the modules a little bit,
+        // and they should live in the IIIF-Storage module.
+        $acl->addResource('iiif-collection');
+        $acl->addResource('iiif-manifest');
+        $acl->addResource('iiif-canvas');
+
+        $acl->allow(
+            null,
+            'iiif-collection',
+            'view-all',
+            new UserHasSitePermission(
+                $settings,
+                'view-all-collections'
+            )
+        );
```

Comment:
This needs some explaining. The role is null, since it applies to all roles, that one's easy. The `iiif-collection` is the resource that the permission is being applied to. It's incorrect to look at this in order to query the ACL however, since `iiif-collection` is a reference to the ID of the resource.

The resource is a wrapper around this ID. In order to have enough information to determine if the user has a site permission we need: the current site, the full user object and the resource being requested. The [wrapper](repos/shared-library/src/Framework/ResourceWrapper.php) accepts the user and site in its constructor.

It is constructed in  the [abstract controller](https://github.com/digirati-co-uk/madoc-platform/pull/131/files#diff-ed379415660c1c012ae213fb1832ca4cR29-R37) so can be accessed by any of the modules. 

When the `UserHasSitePermission` is checked, it takes the user and site from this wrapper and decides if the user can do the thing. It also has an optional resource that we're not using yet. We could pass in the IIIF resource itself and have extended ACLs based on that (For example, lift the rule if you created the IIIF resource under certain configurations). 

Since the wrapper is in the shared module and the IDs are just chosen constants, any module, including custom per-installation modules, can narrow the access to IIIF resources based on the site, user and IIIF module in a decentralised way. 

I think, if this works well, the `ResourceWrapper` could also be a model for other permissions, not specifically IIIF, down the line.